### ISSUE TITLE
Generated destination arguments fail to escape kotlin keywords in package name

### ISSUE THREAD

**Issue by hrafnthor** [raamcosta/compose-destinations#288]
Similarly to #137 the navigation arguments inside of generated `Destination`s fails to sanitize the package names of those arguments

Following are screenshots from the generated `Destination` and the unescaped file paths:

![Screenshot from 2022-11-08 21-23-37](https://user-images.githubusercontent.com/602070/200679635-d8735c5b-33d4-4999-b26e-24ee863a9bcb.png)
![Screenshot from 2022-11-08 21-23-52](https://user-images.githubusercontent.com/602070/200679658-820f8619-a54c-4c67-b026-163145bc478f.png)
![Screenshot from 2022-11-08 21-24-09](https://user-images.githubusercontent.com/602070/200679672-9630197e-c844-4499-84b9-c1f1c7bdd22a.png)

**Comment by hrafnthor** [comment]
I believe this is due to lack of sanitiation on KClass.qualifiedName inside [TypeUtils.kt](https://github.com/raamcosta/compose-destinations/blob/main/compose-destinations-codegen/src/main/java/com/ramcosta/composedestinations/codegen/commons/TypeUtils.kt#L161) here
https://github.com/raamcosta/compose-destinations/blob/0d662f8549184e8ec9372b7a6e652841a24fa62f/compose-destinations-codegen/src/main/java/com/ramcosta/composedestinations/codegen/commons/TypeUtils.kt#L137

and here
https://github.com/raamcosta/compose-destinations/blob/0d662f8549184e8ec9372b7a6e652841a24fa62f/compose-destinations-codegen/src/main/java/com/ramcosta/composedestinations/codegen/commons/TypeUtils.kt#L161

**Comment by raamcosta** [comment]
Hi @hrafnthor 👋 

So sorry it took me so long to get to this.

Can you please give me more information as to how this type `ParcelableApplicationInfo` gets used? What I find weird is that usually it should be imported and then only the `simpleName` would be used here, and since imports are already sanitized, you wouldn't have an issue.
Regardless of this, I know how to reproduce something similar that definitely needs fixing, so thank you!

Another thing I find weird in your screenshots is that only the left side is using qualified name 🤔 
Are we also importing it? Can you check that? If not, how is the second part (after "=") working? I'd expect it to also use the qualified name.

That's why I need more info here, coz it seems like it could be a different issue than the one I can reproduce.

**Comment by hrafnthor** [comment]
I can give more info, sure thing.

The class in question, `ParcelableApplicationInfo` was just a plain `data class` acted as a wrapper for primitive values and was used as a navArgsDelegate for a destionation, as follows:

```kotlin
@Destination(
    navArgsDelegate = ParcelableApplicationInfo::class
)
fun ListScreen() { /*...*/ }
```

As the package for  `ParcelableApplicationInfo` contains the Kotlin keyword `is`, the class generated by Compose-Destination contained the unsanitized package name inline where `ParcelableApplicationInfo` was used as seen in the screenshots above (however due to a previous fix #284 the import location had the correct sanitized package name).

I'm not sure what could have been causing the full qualified package name to be loaded into the generated class file, as my setup is as vanilla as could be. The only thing I can think of (and even then I'm doubtful) is that this is happening inside of a library module which has the following compile flag turned on `-Xexplicit-api=strict` to enforce explicit visibility definitions.

Due to this issue being a hard blocker, and time constraints, I had to move away from using Compose Destinations in the library in question, and so do currently not have the setup available to reproduce the issue again. At the time though, the issue got constantly regenerated with the full unsanitized package name.

**Comment by raamcosta** [comment]
From what I can see in the screenshots, this class was not considered as the NavArhs delegate class. It was considered as a normal nav argument.

I still don't see how this happens that the type itself shows as qualified name but not the instantiation for the default argument 🤷 

I think my fix should still cover this, and since you can't reproduce it easily, I'll close this. Let me know if you have a chance of trying compose destinations again and this is fixed 🙂
